src/gui/EditWidgetIcons.cpp

```diff
@@ -314,14 +314,15 @@ void EditWidgetIcons::addCustomIconFromFile()
         QString filter = QString("%1 (%2);;%3 (*)").arg(tr("Images"),
                     Tools::imageReaderFilter(), tr("All files"));
 
-        QString filename = QFileDialog::getOpenFileName(
-                    this, tr("Select Image"), "", filter);
-        if (!filename.isEmpty()) {
-            auto icon = QImage(filename);
-            if (!icon.isNull()) {
-                addCustomIcon(QImage(filename));
-            } else {
-                emit messageEditEntry(tr("Can't read icon"), MessageWidget::Error);
+        auto filenames = QFileDialog::getOpenFileNames(this, tr("Select Image(s)"), "", filter);
+        for (const auto& filename : filenames) {
+            if (!filename.isEmpty()) {
+                auto icon = QImage(filename);
+                if (!icon.isNull()) {
+                    addCustomIcon(QImage(filename));
+                } else {
+                    emit messageEditEntry(tr("Can't read icon"), MessageWidget::Error);
```

Comment:
An improvement: report in the message the filename that can't be loaded.

Also if I load 100 corrupted icons this message will be "spammed" 100 times.
IMHO this needs to be reported at the end of the for cycle with a list of the error filenames.
If the list is too long display only the first `N` filename

Reply:
This was way more complicated than it needed to be, but after some replumbing it woks great!